release-common-lib/src/main/java/org/reactome/util/compare/DBObjectComparer.java

```diff
@@ -287,317 +474,125 @@ private static String getInstanceTypeMismatchMessage(GKInstance instance1, GKIns
-	 * @param instance Instance for which to get referrer attributes
-	 * @return List of referrer SchemaAttributes of the instance passed
-	 */
-	@SuppressWarnings("unchecked")
-	private static List<SchemaAttribute> getReferrerAttributes(GKInstance instance)
-	{
-		List<SchemaAttribute> referrerAttributes = instanceToReferrerAttributesMap.get(instance);
-		if (referrerAttributes == null)
-		{
-			referrerAttributes = new ArrayList<SchemaAttribute>(instance.getSchemClass().getReferers());
-			instanceToReferrerAttributesMap.put(instance, referrerAttributes);
-		}
-		return referrerAttributes;
-	}
-
-	/**
-	 * Filters the passed list of attributes based on the predicate function passed and returns the filtered list.
-	 * @param attributes List of attributes to filter
-	 * @param attributeNameFilter Predicate value used to determine which values should be included
-	 * @return List of SchemaAttributes after filtering
-	 */
-	private static List<SchemaAttribute> filterAttributes(
-		List<SchemaAttribute> attributes, Predicate<? super SchemaAttribute> attributeNameFilter
+		SchemaAttribute attribute, AttributeRelationshipType attributeRelationship, GKInstance instance1,
+		GKInstance instance2, int values1Size, int values2Size
 	)
 	{
-		return attributes
-			.stream()
-			.filter(attributeNameFilter != null ? attributeNameFilter : getDefaultAttributeNameFilter())
-			.collect(Collectors.toList());
+		return "Count mismatch for " + attributeRelationship +
+			" '" + attribute.getName() + "'" + System.lineSeparator() +
+			" Instance 1 ('" + instance1 + "' from " + getDBName(instance1) + ") has " +
+			values1Size + " elements " + System.lineSeparator() +
+			" Instance 2 ('" + instance2 + "' from " + getDBName(instance2) + ") has " +
+			values2Size + " elements" + System.lineSeparator() + System.lineSeparator();
 	}
 
 	/**
-	 * Returns the default attribute filter as a Predicate function.  By default, the attributes "DB_ID", "dateTime",
-	 * "modified", and "created" are filtered out (i.e. ignored).
-	 * @return Predicate function used to filter out SchemaAttributes value by their names
+	 * Returns the name of the database from which the passed instance originates.
+	 * @param instance Instance for which to get the name of the database from which it originated
+	 * @return Name of the database of origin for the passed instance
 	 */
-	private static Predicate<? super SchemaAttribute> getDefaultAttributeNameFilter()
+	private static String getDBName(GKInstance instance)
 	{
-		/*
-		 * Used for filtering out attributes that you don't want to compare because they will
-		 * probably create too much noise if you do.
-		 */
-		return a -> {
-			return !a.getName().equals(ReactomeJavaConstants.DB_ID)
-				&& !a.getName().equals(ReactomeJavaConstants.dateTime)
-				&& !a.getName().equals(ReactomeJavaConstants.modified)
-				&& !a.getName().equals(ReactomeJavaConstants.created);
-		};
+		return ((MySQLAdaptor) instance.getDbAdaptor()).getDBName();
```

Comment:
Funny that you need to cast `MySQLAdaptor` for the method `getDbAdaptor()`

Reply:
Yeah, `getDbAdaptor()` returns the interface type `PersistenceAdaptor` which is implemented by `MySQLAdaptor` and a few other things, but when working in this context the actual object is always a `MySQLAdaptor`.